### ISSUE TITLE
QuoteRepository get methods won't return CartInterface but Quote model

### ISSUE THREAD

**Issue by msieprawski** [magento/magento2#12802]
### Preconditions
1. Magento 2.2.0 CE
2. PHP 7.024

### Steps to reproduce
1. Create custom module with following etc/di.xml` file:
```
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <preference for="Magento\Quote\Api\Data\CartInterface" type="Vendor\Module\Model\Quote" />
</config>
```
2. Create `\Vendor\Module\Model\Quote` object which extends `\Magento\Quote\Model\Quote`.
3. Now execute following code:
```
$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
$cartRepository = $objectManager->create(\Magento\Quote\Api\CartRepositoryInterface::class);
$cartRepository->get(70); // use existing quote ID
```
### Expected result
1. The result will be an object of instance `Vendor\Module\Model\Quote`

### Actual result
1. Actual result is an object of instance `Magento\Quote\Model\Quote`

It's caused by invalid vendor/magento/module-quote/Model/QuoteRepository.php::loadQuote method which uses `Magento\Quote\Model\QuoteFactory` in the very first line instead of `Magento\Quote\Api\Data\CartInterfaceFactory`.

The solution is to override QuoteRepository's loadQuote method and use CartInterfaceFactory.

**Comment by msieprawski** [comment]
Any updates on this one?

**Comment by magento-engcom-team** [comment]
@msieprawski, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by cpjeanpaul** [comment]
So... ?

**Comment by magento-engcom-team** [comment]
Hi @msieprawski. Thank you for your report. 
The issue has been fixed in magento/magento2#22149 by @Bartlomiejsz in 2.3-develop branch
Related commit(s):
 - [865d7df16d374246d91d35f1bdb737828c88edc8](https://github.com/magento/magento2/commit/865d7df16d374246d91d35f1bdb737828c88edc8)
 - [6e079a341326aa7bf3a59848cbadbf2153eb4ae0](https://github.com/magento/magento2/commit/6e079a341326aa7bf3a59848cbadbf2153eb4ae0)
 - [fa5cf907e0bd9f0fda4431a551e798a911dfe875](https://github.com/magento/magento2/commit/fa5cf907e0bd9f0fda4431a551e798a911dfe875)

The fix will be available with the upcoming 2.3.2 release.

**Comment by sidolov** [comment]
Hi @msieprawski. Thank you for your report. 
The issue has been fixed in magento/magento2#22549 by @Bartlomiejsz in 2.2-develop branch
Related commit(s):
 - [2a148f665eac3bcebf31110a456e373583946cd6](https://github.com/magento/magento2/commit/2a148f665eac3bcebf31110a456e373583946cd6)
 - [ac0e9330ee930073fcc76b3345fc86cf3976193a](https://github.com/magento/magento2/commit/ac0e9330ee930073fcc76b3345fc86cf3976193a)

The fix will be available with the upcoming 2.2.10 release.
